### PR TITLE
[sui] Simplify some tests and add more

### PR DESCRIPTION
**Pull request by m30m**

### CODE REVIEW

File: target_chains/sui/contracts/sources/pyth.move [hunk 1]
```diff
@@ -953,52 +926,158 @@ module pyth::pyth_tests{
...
+
+     #[test]
+    #[expected_failure(abort_code = pyth::accumulator::E_INVALID_PROOF)]
+    fun test_accumulator_invalid_proof() {
```
Comment (jayantk) [inline target_chains/sui/contracts/sources/pyth.move:942]: can these be cut down even more in terms of code duplication? it seems like the key variables are the setup and the message bytes, and then there's quite a bit of additional boilerplate.

**Review by jayantk — APPROVED** [review]
nice

File: target_chains/sui/contracts/sources/pyth.move [hunk 2]
```diff
@@ -1043,57 +1145,30 @@ module pyth::pyth_tests{
...
-        let price_info_1 = price_info::get_price_info_from_price_info_object(&price_info_object_1);
-        assert!(price_feeds_equal(&price_info_1, vector::borrow(&expected_price_infos, 0)), 0);
-
-        let price_info_2 = price_info::get_price_info_from_price_info_object(&price_info_object_2);
-        assert!(price_feeds_equal(&price_info_2, vector::borrow(&expected_price_infos, 1)), 0);
-
-        let price_info_3 = price_info::get_price_info_from_price_info_object(&price_info_object_3);
-        assert!(price_feeds_equal(&price_info_3, vector::borrow(&expected_price_infos, 2)), 0);
-
+        while (idx < 3){
```
Comment (optke3) [inline target_chains/sui/contracts/sources/pyth.move:1151]: nice simplification

**Review by optke3 — APPROVED** [review]
LGTM

File: target_chains/sui/contracts/sources/pyth.move (hunk 1 above)
Comment (m30m) [reply target_chains/sui/contracts/sources/pyth.move:942]: Yes, will do another compression in the end in a separate PR

**Pull request merged**